src/t_zset.c

```diff
@@ -2912,19 +2923,22 @@ static void zrangeResultEmitLongLongToClient(zrange_result_handler *handler,
 static void zrangeResultFinalizeClient(zrange_result_handler *handler,
     size_t result_count)
 {
-    /* In case of WITHSCORES, respond with a single array in RESP2, and
-     * nested arrays in RESP3. We can't use a map response type since the
-     * client library needs to know to respect the order. */
-    if (handler->withscores && (handler->client->resp == 2)) {
-        result_count *= 2;
-    }
+    if (handler->userdata) {
+        /* In case of WITHSCORES, respond with a single array in RESP2, and
+        * nested arrays in RESP3. We can't use a map response type since the
+        * client library needs to know to respect the order. */
+        if (handler->withscores && (handler->client->resp == 2)) {
+            result_count *= 2;
+        }
 
-    setDeferredArrayLen(handler->client, handler->userdata, result_count);
+        setDeferredArrayLen(handler->client, handler->userdata, result_count);
+    }
```

Comment:
maybe use an early return (with some comment), instead of indenting all this code?